colossalai/context/parallel_context.py

```diff
@@ -81,6 +84,13 @@ def load_config(self, config: Union[dict, str]):
         else:
             raise TypeError("Invalid type for config, only dictionary or string is supported")
 
+    def detect_num_processes_on_current_node(self):
```

Comment:
Can we just run this once and cache the results. 
Frequent all gather is not good for performance

Reply:
Yes, it only runs once in colossalai.launch